### ISSUE TITLE
Alpine 3.17 mold sometimes timeouts the CI

### ISSUE THREAD

**Issue by okias** [rui314/mold#898]
- Dist: Alpine 3.17
- mold 1.7.1-r0
- arch x86_64

Mold freezing without error time to time (ratio ~ 10:1).

Currently all I can provide, see the https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/20032

Docker: registry.freedesktop.org/mesa/mesa/alpine/x86_build:2022-11-23-alpine-3_17--d5aa3941aa03c2f716595116354fb81eb8012acb

**Comment by rui314** [comment]
- Was your mold executable built with the bundled onetbb, or with a system onetbb with my patch (https://github.com/rui314/mold/commit/d8dd12426c1f20c21dfcf5d4021a7c89c88ed727)?
- Did it happen before 1.7.0? What was the last version that worked stably?

**Comment by okias** [comment]
- build log: https://build.alpinelinux.org/buildlogs/build-3-17-armv7/community/mold/mold-1.7.1-r0.log
- build options: https://git.alpinelinux.org/aports/tree/community/mold/APKBUILD?h=3.17-stable
  ```
	cmake -B build -G Ninja \
		-DCMAKE_INSTALL_PREFIX=/usr \
		-DCMAKE_BUILD_TYPE=Release \
		-DMOLD_LTO=ON \
		-DMOLD_USE_SYSTEM_MIMALLOC=ON \
		-DMOLD_USE_SYSTEM_TBB=ON \
		-DBUILD_TESTING="$(want_check && echo ON || echo OFF)"
  ```
- no patches

Mold got recently packaged for Alpine (`3.17` and `edge`) and I somehow couldn't build 1.6.x on Alpine `3.16`, so I didn't tested.

**Comment by rui314** [comment]
Since it was built with `-DMOLD_USE_SYSTEM_TBB=ON`, it uses the system's onetbb and not the bundled one. What if you remove that option, rebuild mold and try again?

**Comment by okias** [comment]
Thank you! I'll look into it, but cannot promise it'll be today :wink: (since it's distribution package which I'll have to setup locally from the CI docker container and build it)

**Comment by ishitatsuyuki** [comment]
Thanks for the report.

I actually noticed the revert on Mesa, and I tried running the CI docker container from the failing revision, with the exact `meson` command provided in the CI image. However, the build completed without any issues here.

I wonder it's just some swap stuff in play? mold might be slightly more memory hungry than traditional linkers due to its aggressive and parallel loading of input files.
